### PR TITLE
Writing log readme in multiprocess safe way

### PR DESCRIPTION
**Pull request by mwojtyczka**
## Changes

1. Changed writing of the readme log in a multiprocess safe way. Without this the assessment tasks are occasionally failing with error: `FileExistsError: [Errno 17] File exists: '/Workspace/Users/jni4fe@xxx.com/.ucx/logs/assessment/run-340934288146817/README.md'`
2. Skip long running integration test for Redash
3. Check if job settings and tasks are present while crawling

### Tests
- [x] manually tested

### CODE REVIEW

File: src/databricks/labs/ucx/framework/tasks.py [hunk 1]
```diff
@@ -116,6 +120,42 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+@contextmanager
+def exclusive_open(filename: str, *args, timeout: int = 5, retry_time: float = 0.05, **kwargs):
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/tasks.py:124]: use `timedelta(seconds=5)` for timeout

File: src/databricks/labs/ucx/framework/tasks.py [hunk 2]
```diff
@@ -116,6 +120,42 @@ def wrapper(*args, **kwargs):
...
+    """
+    lockfile = filename + ".lock"
+    deadline = datetime.now() + timedelta(seconds=timeout)
+
+    # wait until the lock file can be opened
+    while True:
+        try:
+            fd = os.open(lockfile, os.O_CREAT | os.O_EXCL)
+            break
+        except FileExistsError:
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/tasks.py:143]: use `@retried(on=[FileExistsError], timeout=...)`

File: src/databricks/labs/ucx/framework/tasks.py [hunk 3]
```diff
@@ -168,8 +208,8 @@ def trigger(*argv):
 
     log_readme = log_path.joinpath("README.md")
     if not log_readme.exists():
-        # this may race when run from multiple tasks, but let's accept the risk for now.
-        with log_readme.open(mode="w") as f:
+        # this may race when run from multiple tasks, therefore it must be multiprocess safe
+        with exclusive_open(str(log_readme), mode="w") as f:
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/tasks.py:212]: is there a method for this on `pathlib.Path`?

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/framework/tasks.py (hunk 3 above)
Comment (mwojtyczka) [reply src/databricks/labs/ucx/framework/tasks.py:212]: do you mean to get the file as string?

File: src/databricks/labs/ucx/framework/tasks.py (hunk 1 above)
Comment (mwojtyczka) [reply src/databricks/labs/ucx/framework/tasks.py]: corrected

File: src/databricks/labs/ucx/framework/tasks.py (hunk 2 above)
Comment (mwojtyczka) [reply src/databricks/labs/ucx/framework/tasks.py]: corrected

**Pull request merged**